### DISCUSSION TITLE
Add support for create function, class or parameter code actions depending on context

### DISCUSSION THREAD

**Discussion by luabud** [microsoft/pylance-release#3600] [Ideas]
It'd be nice Pylance could displayed refactoring code actions depending on the context. For example:

```py

def foo():
    return x  # <--- 💡Create parameter x

a = MyClass # <--- 💡 Create class MyClass

b = bla() # <---- 💡 Create function bla()



```

**Comment by luabud** [comment]
Related to https://github.com/microsoft/vscode-python/issues/2653

**Comment by MartinBernstorff** [comment]
This would be an enormous improvement on my workflow; any news here? :-)

**Comment by judej** [comment]
Thanks for input in this discussion. We have added this item to our backlog. Check this #5368 issue to track progress. 
thanks!
